File: frontend-app/src/models/payment.ts
import { Benefactor, Client, GetOptions, Validation } from ".";

export interface Payment {
    id?: number;
    comment: string;
    amount: number;

    status: PaymentStatuses

    benefactor?: Benefactor;
    client?: Client;
    request?: Request;
}

export enum PaymentStatuses {
    InProgress,
    Completed,
    Error
}

export namespace Payment {
    export const initial: Payment = {
        amount: 0,
        comment: '',
        status: PaymentStatuses.InProgress
    };
}

export interface PaymentGetOptions extends GetOptions {
    benefactorId?: number;
    requestId?: number;
}
export interface PaymentValidation extends Validation {
    benefactorError?: string;
}
export namespace PaymentValidation {
    export const initial: PaymentValidation = Validation.initial;
}

File: back-end/Hie.DB.Entites/Payment.cs
using Hie.DB.Common;
using System;

namespace Hie.DB.Entities {
  public class Payment: IEntity, ICreatedDate {
    public long Id { get; set; }
    public DateTime CreateDateUtc { get; set; }
    public decimal Amount { get; set; }
    public string Comment { get; set; }

    public long BenefactorId { get; set; }
    public Benefactor Benefactor { get; set; }

    public long? RequestId { get; set; }
    public Request Request { get; set; }
  }
}